fenix/app/src/main/java/org/mozilla/fenix/translations/TranslationsDialogBottomSheet.kt

```diff
@@ -262,43 +320,130 @@ private fun TranslationsDialogInfoMessage(
     }
 }
 
+@Suppress("LongMethod")
 @Composable
 private fun TranslationsDropdown(
     header: String,
+    translateLanguages: List<Language>,
     modifier: Modifier = Modifier,
+    isInLandscapeMode: Boolean = false,
 ) {
-    Column(modifier, verticalArrangement = Arrangement.spacedBy(4.dp)) {
+    val density = LocalDensity.current
+
+    var expanded by remember { mutableStateOf(false) }
+
+    var selectedLanguage by remember {
+        mutableStateOf(
+            translateLanguages.last().localizedDisplayName,
+        )
+    }
+
+    var contextMenuWidthDp by remember {
+        mutableStateOf(0.dp)
+    }
+
+    Column(
+        modifier = modifier
+            .clickable {
+                expanded = true
+            }
+            .semantics { role = Role.DropdownList },
+    ) {
         Text(
             text = header,
+            modifier = Modifier.wrapContentSize(),
             color = FirefoxTheme.colors.textPrimary,
             style = FirefoxTheme.typography.caption,
         )
 
+        Spacer(modifier = Modifier.height(4.dp))
+
         Row {
-            Text(
-                text = "English",
-                modifier = Modifier.weight(1f),
-                color = FirefoxTheme.colors.textPrimary,
-                style = FirefoxTheme.typography.subtitle1,
-            )
+            selectedLanguage?.let {
+                Text(
+                    text = it,
+                    modifier = Modifier.weight(1f),
+                    color = FirefoxTheme.colors.textPrimary,
+                    style = FirefoxTheme.typography.subtitle1,
+                )
+            }
 
             Spacer(modifier = Modifier.width(10.dp))
 
-            Icon(
-                painter = painterResource(id = R.drawable.mozac_ic_dropdown_arrow),
-                contentDescription = null,
-                tint = FirefoxTheme.colors.iconPrimary,
-            )
+            Box {
+                Icon(
+                    painter = painterResource(id = R.drawable.mozac_ic_dropdown_arrow),
+                    contentDescription = null,
+                    modifier = if (expanded) {
+                        Modifier.rotate(EXPANDED_ICON_DEGREES)
+                    } else {
+                        Modifier.rotate(
+                            0f,
+                        )
+                    },
```

Comment:
Is it expected for the icon to rotate? I feel like this could just be static.

Reply:
Done , I removed it.